Backpack/Theme/Classes/BPKContainerController.m

```diff
@@ -0,0 +1,178 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "BPKContainerController.h"
+#import "BPKTheme.h"
+
+#import <Backpack/Common.h>
+
+NS_ASSUME_NONNULL_BEGIN
+@interface BPKContainerController ()
+@property(nonatomic, strong) UIViewController *rootViewController;
+@end
+
+@implementation BPKContainerController
+
+- (void)dealloc {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (instancetype)initWithThemeContainer:(UIView *)themeContainer
+                    rootViewController:(nonnull UIViewController *)rootViewController {
+    self = [super initWithNibName:nil bundle:nil];
+
+    if (self) {
+        _themeActive = YES;
+        _themeContainer = themeContainer;
+        self.rootViewController = rootViewController;
+        
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(receiveContainer:)
+                                                     name:BPKTheme.didChangeContainerNotification
+                                                   object:nil];
+    }
+
+    return self;
+}
+
+- (void)receiveContainer:(NSNotification *)notification {
```

Comment:
Doesn't this logic belong in the subclass that knows about theming?

Reply:
Well this is just recieving  any UIView class for creating a container. It's not specific to theming. Could provide any view (theme container or not)